### PR TITLE
refactor: PlaceholderFile own user data

### PR DESCRIPTION
**Pull request by ho-229**
This PR is refactors `PlaceholderFile` to own user data instead of borrowing it. It will make `PlaceholderFile` more easy to use.

### CODE REVIEW

**Comment by Xuanwo** [comment]
Hi, @ok-nick would you like to take a look at this PR?

**Review by ok-nick — CHANGES_REQUESTED** [review]

**Comment by ok-nick** [comment]
@Xuanwo Hey, just realized my review was pending for over a week. Submitted it now.

File: src/command/commands.rs [hunk 1]
```diff
@@ -54,7 +54,7 @@ impl Command for Read<'_> {
 }
 
 /// Write data to a placeholder file.
-#[derive(Debug, Clone, Default)]
```
Comment (ok-nick) [inline src/command/commands.rs:57]: Why did you remove Clone/Default derives from all commands?

File: src/placeholder.rs [hunk 2]
```diff
@@ -259,13 +259,17 @@ impl Seek for Placeholder {
...
 
 impl<'a> UpdateOptions<'a> {
     /// Create a new [UpdateOptions][crate::UpdateOptions].
     pub fn new() -> Self {
-        Self::default()
+        Self(Update {
+            mark_sync: false,
+            metadata: None,
+            blob: None,
+        })
```
Comment (ok-nick) [inline src/placeholder.rs:272]: How come you removed the default impl?

**Review by ok-nick — CHANGES_REQUESTED** [review]

File: src/command/commands.rs (hunk 1 above)
Comment (ho-229) [reply src/command/commands.rs:57]: I think commands should not be cloned, and parameters must be specified explicitly.

**Review by ok-nick — APPROVED** [review]
